src/ooo_cpu.cc

```diff
@@ -1563,178 +1601,204 @@ void O3_CPU::reg_RAW_release(uint32_t rob_index)
 
 void O3_CPU::operate_cache()
 {
-    ITLB.operate();
-    DTLB.operate();
-    STLB.operate();
-    L1I.operate();
-    L1D.operate();
-    L2C.operate();
+    ITLB.operate_writes();
+    DTLB.operate_writes();
+    STLB.operate_writes();
+    L1I.operate_writes();
+    L1D.operate_writes();
+    L2C.operate_writes();
+
+    L2C.operate_reads();
+    L1D.operate_reads();
+    L1I.operate_reads();
+    STLB.operate_reads();
+    DTLB.operate_reads();
+    ITLB.operate_reads();
```

Comment:
I'm not certain what the purpose here is to have the writes propagate in one direction and the reads propagate in another. Why will writes be able to shoot through, but not reads?

Reply:
It's more important for performance realism reasons to prevent fills from shooting through on their way from DRAM to the L1 caches.  This makes it so the transition between each level in both directions takes at least 1 cycle.

Reply:
This makes me wonder if it's better to fix shootthrough using the `event_cycle` members, setting it to a later cycle. That seems like it could fix the problem. I noted in #63 that this is already not handled well.